### ISSUE TITLE
Weecord fails to compile

### ISSUE THREAD

**Issue by ryzokuken** [khyperia/weechat-discord#24]
Here's the log:

```
   Compiling weecord v0.1.0 (file:///home/ryzokuken/Software/weechat-discord)
error: failed to run custom build command for `weecord v0.1.0 (file:///home/ryzokuken/Software/weechat-discord)`
process didn't exit successfully: `/home/ryzokuken/Software/weechat-discord/target/release/build/weecord-5d4e49d62e5557be/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("release")
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/usr/include/weechat" "-Wall" "-Wextra" "-Werror" "-o" "/home/ryzokuken/Software/weechat-discord/target/release/build/weecord-b7f74f9e6f476cee/out/src/weecord.o" "-c" "src/weecord.c"
cargo:warning=src/weecord.c: In function ‘wdc_hook_command’:
cargo:warning=src/weecord.c:60:35: error: macro "weechat_hook_command" passed 8 arguments, but takes just 7
cargo:warning=                               NULL);
cargo:warning=                                   ^
cargo:warning=src/weecord.c:53:10: error: ‘weechat_hook_command’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_command(command,
cargo:warning=          ^
cargo:warning=src/weecord.c:53:10: note: each undeclared identifier is reported only once for each function it appears in
cargo:warning=src/weecord.c:40:30: error: unused parameter ‘command’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_command(const char* command,
cargo:warning=                              ^
cargo:warning=src/weecord.c:41:30: error: unused parameter ‘description’ [-Werror=unused-parameter]
cargo:warning=                  const char* description,
cargo:warning=                              ^
cargo:warning=src/weecord.c:42:30: error: unused parameter ‘args’ [-Werror=unused-parameter]
cargo:warning=                  const char* args,
cargo:warning=                              ^
cargo:warning=src/weecord.c:43:30: error: unused parameter ‘args_description’ [-Werror=unused-parameter]
cargo:warning=                  const char* args_description,
cargo:warning=                              ^
cargo:warning=src/weecord.c:44:30: error: unused parameter ‘completion’ [-Werror=unused-parameter]
cargo:warning=                  const char* completion,
cargo:warning=                              ^
cargo:warning=src/weecord.c:45:30: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=                  const void* pointer,
cargo:warning=                              ^
cargo:warning=src/weecord.c:46:24: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=                  int (*callback)(const void* pointer,
cargo:warning=                        ^
cargo:warning=src/weecord.c: In function ‘wdc_buffer_new’:
cargo:warning=src/weecord.c:117:71: error: macro "weechat_buffer_new" passed 7 arguments, but takes just 5
cargo:warning=     name, input_callback, pointer, NULL, close_callback, pointer, NULL);
cargo:warning=                                                                       ^
cargo:warning=src/weecord.c:116:10: error: ‘weechat_buffer_new’ undeclared (first use in this function)
cargo:warning=   return weechat_buffer_new(
cargo:warning=          ^
cargo:warning=src/weecord.c:106:28: error: unused parameter ‘name’ [-Werror=unused-parameter]
cargo:warning= wdc_buffer_new(const char* name,
cargo:warning=                            ^
cargo:warning=src/weecord.c:107:28: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=                const void* pointer,
cargo:warning=                            ^
cargo:warning=src/weecord.c:108:22: error: unused parameter ‘input_callback’ [-Werror=unused-parameter]
cargo:warning=                int (*input_callback)(const void* pointer,
cargo:warning=                      ^
cargo:warning=src/weecord.c:112:22: error: unused parameter ‘close_callback’ [-Werror=unused-parameter]
cargo:warning=                int (*close_callback)(const void* pointer,
cargo:warning=                      ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_fd’:
cargo:warning=src/weecord.c:147:62: error: macro "weechat_hook_fd" passed 7 arguments, but takes just 6
cargo:warning=   return weechat_hook_fd(fd, 1, 0, 0, callback, pointer, NULL);
cargo:warning=                                                              ^
cargo:warning=src/weecord.c:147:10: error: ‘weechat_hook_fd’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_fd(fd, 1, 0, 0, callback, pointer, NULL);
cargo:warning=          ^
cargo:warning=src/weecord.c:143:17: error: unused parameter ‘fd’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_fd(int fd,
cargo:warning=                 ^
cargo:warning=src/weecord.c:144:25: error: unused parameter ‘pointer’ [-Werror=unused-parameter]
cargo:warning=             const void* pointer,
cargo:warning=                         ^
cargo:warning=src/weecord.c:145:19: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=             int (*callback)(const void* pointer, void* data, int fd))
cargo:warning=                   ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_completion’:
cargo:warning=src/weecord.c:231:67: error: macro "weechat_hook_completion" passed 5 arguments, but takes just 4
cargo:warning=     completion_item, description, callback, callback_pointer, NULL);
cargo:warning=                                                                   ^
cargo:warning=src/weecord.c:230:10: error: ‘weechat_hook_completion’ undeclared (first use in this function)
cargo:warning=   return weechat_hook_completion(
cargo:warning=          ^
cargo:warning=src/weecord.c:221:33: error: unused parameter ‘completion_item’ [-Werror=unused-parameter]
cargo:warning= wdc_hook_completion(const char* completion_item,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:222:33: error: unused parameter ‘description’ [-Werror=unused-parameter]
cargo:warning=                     const char* description,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:223:33: error: unused parameter ‘callback_pointer’ [-Werror=unused-parameter]
cargo:warning=                     const void* callback_pointer,
cargo:warning=                                 ^
cargo:warning=src/weecord.c:224:27: error: unused parameter ‘callback’ [-Werror=unused-parameter]
cargo:warning=                     int (*callback)(const void*,
cargo:warning=                           ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_command’:
cargo:warning=src/weecord.c:61:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_buffer_new’:
cargo:warning=src/weecord.c:118:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_fd’:
cargo:warning=src/weecord.c:148:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=src/weecord.c: In function ‘wdc_hook_completion’:
cargo:warning=src/weecord.c:232:1: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning= }
cargo:warning= ^
cargo:warning=cc1: all warnings being treated as errors
exit code: 1


command did not execute successfully, got: exit code: 1



--- stderr
thread 'main' panicked at 'explicit panic', /home/ryzokuken/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.50/src/lib.rs:1210
note: Run with `RUST_BACKTRACE=1` for a backtrace.

Build failed, waiting for other jobs to finish...
error: build failed
Makefile:17: recipe for target 'all' failed
make: *** [all] Error 101
```

**Comment by khyperia** [comment]
Duplicate of #23. Please see the resolution there.
